docs/single-page-app.md

```diff
@@ -0,0 +1,213 @@
+# Single page app
+
+The `single-page-app` construct deploys:
+
+- **single-page applications**, for example React or VueJS applications
+
+## Quick start
+
+```bash
+serverless plugin install -n serverless-lift
+```
+
+```yaml
+service: my-app
+provider:
+    name: aws
+
+constructs:
+    landing:
+        type: single-page-app
+        path: public
+
+plugins:
+    - serverless-lift
+```
+
+On `serverless deploy`, the `public/` directory will be deployed as a public website.
+
+_Note: **the first deployment takes 4 minutes**. Next deployments only take seconds._
+
+The website is served over HTTPS and cached all over the world via the CloudFront CDN.
+
+## How it works
+
+On the first `serverless deploy`, Lift creates:
+
+- an [S3](https://aws.amazon.com/s3/) bucket
+- a [CloudFront CDN](https://aws.amazon.com/cloudfront/) configured to serve the website from S3 over HTTPS, with caching at the edge
+- CloudFront Functions to set security HTTP headers
+
+![](img/static-website.png)
+
+Additionally, every time `serverless deploy` runs, Lift:
+
+- uploads all files from the configured directory to the S3 bucket
+- invalidates the CloudFront cache so that the new version of the website is live
+
+To learn more about the architecture of this construct, [read this article](https://medium.com/serverless-transformation/static-websites-on-aws-designing-lift-1db94574ba3b).
+
+_Note: the S3 bucket is public and entirely managed by Lift. Do not store or upload files to the bucket, they will be removed by Lift on the next deployment. Instead, create a separate bucket to store any extra file._
```

Comment:
Do you still require the S3 bucket to be public with the Cloudfront function construct rebuilding URL to match `index.html` ? This is a security concern for some companies to not allow S3 content to be accessible bypassing Cloudfront alltogether

Reply:
If we can skip making the bucket public that would be a huge win indeed.

Reply:
Done !